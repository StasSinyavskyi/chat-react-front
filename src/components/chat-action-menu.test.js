/* eslint-env jest */
import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';
import ChatActionMenu from './chat-action-menu';

// if we have several test we can join them in describe
describe('<ChatActionMenu />', () => {
  const testParam = (
    <ChatActionMenu
      onLiveClick={() => 'onLiveClick'}
      onDeleteClick={() => 'onDeleteClick'}
      activeUser={{ isChatMember: true, isCreator: true }}
      disabled={false}
    />
  );

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(testParam, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('render correctly', () => {
    const tree = renderer
      .create(testParam)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
