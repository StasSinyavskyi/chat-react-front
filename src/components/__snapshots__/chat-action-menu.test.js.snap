// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatActionMenu /> render correctly 1`] = `
<div>
  <button
    aria-haspopup="true"
    aria-owns={null}
    className="MuiButtonBase-root-46 MuiButton-root-31 ChatActionMenu-button-30"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label-32"
    >
      Action
    </span>
    <span
      className="MuiTouchRipple-root-48"
    />
  </button>
</div>
`;
